src/Common/Commands.Common/CustomAttributes/GenericBreakingChangeAttribute.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            _message = message;
+            this.DeprecateByVersion = deprecateByVersion;
+            this.DeprecateByVersionSet = true;
+
+            this.ChangeInEfectByDate = DateTime.Parse(changeInEfectByDate);
+            this.ChangeInEfectByDateSet = true;
+        }
+
+        public DateTime getInEffectByDate()
+        {
+            return this.ChangeInEfectByDate.Date;
+        }
+
+        /**
+         * This function returns the breaking change text for the attribute
+         * If the withCmdletName is true we return the message with the cmdlet name in it otherwse not
+         *
+         * We get the cmdlet name from the passed in Type (it is expected to have the CMdlet attribute decorated on the class)
+         */
+        public string GetBreakingChangeTextFromAttribute(Type type, bool withCmdletName)
+        {
+            string breakingChangeMessage = null;
+
+            if (!withCmdletName)
+            {
+                breakingChangeMessage = string.Format(Resources.BreakingChangesAttributesDeclarationMessage, GetAttributeSpecificMessage());
+            } else
+            {
+                
+                breakingChangeMessage = string.Format(Resources.BreakingChangesAttributesDeclarationMessageWithCmdletName, Utilities.GetNameFromCmdletType(type), GetAttributeSpecificMessage());
+            }
+
+            if (!string.IsNullOrWhiteSpace(ChangeDescription))
+            {
+                breakingChangeMessage += string.Format(Resources.BreakingChangesAttributesChangeDescriptionMessage, this.ChangeDescription);
+            }
+
+            if (ChangeInEfectByDateSet)
+            {
+                breakingChangeMessage += string.Format(Resources.BreakingChangesAttributesInEffectByDateMessage, this.ChangeInEfectByDate);
+            }
+
+            if (DeprecateByVersionSet)
+            {
+                breakingChangeMessage += string.Format(Resources.BreakingChangesAttributesInEffectByVersion, this.DeprecateByVersion);
+            }
+
+            if (!string.IsNullOrWhiteSpace(OldWay) && !string.IsNullOrWhiteSpace(NewWay))
+            {
+                breakingChangeMessage += string.Format(Resources.BreakingChangesAttributesUsageChangeMessage, OldWay, NewWay);
+            }
+
+            return breakingChangeMessage;
+        }
+
+         /**
+         * This function prints out the breaking change message for the attribute on the cmdline
+         * If the "withCmdletName" is specified, the message is printed out with the cmdlet name in it
+         * otherwise not
+         * 
+         * We get the cmdlet name from the passed in Type (it is expected to have the CMdlet attribute decorated on the class)
+         * */
+        public void PrintCustomAttributeInfo(Type type, bool withCmdletName)
+        {
+            if (!withCmdletName) {
+                Console.WriteLine(string.Format(Resources.BreakingChangesAttributesDeclarationMessage, GetAttributeSpecificMessage()));
```

Comment:
We should not use Console.WriteLine.  In PowerShell, these will go out to the warning stream, meaning WriteWarning.  So we will need to pass a method that wrtes to the appropriate stream using the cmdlets ICommandRuntime so we can write directly.